pkg/operator/controllers/machineset/machineset_controller.go

```diff
@@ -0,0 +1,95 @@
+package machineset
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+type MachineSetReconciler struct {
+	log    *logrus.Entry
+	maocli maoclient.Interface
+	arocli aroclient.Interface
+}
+
+func NewMachineSetReconciler(log *logrus.Entry, maocli maoclient.Interface, arocli aroclient.Interface) *MachineSetReconciler {
+	return &MachineSetReconciler{
+		log:    log,
+		maocli: maocli,
+		arocli: arocli,
+	}
+}
+
+func (r *MachineSetReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	machinesetObject, err := r.maocli.MachineV1beta1().MachineSets(machineSetsNamespace).Get(ctx, request.Name, metav1.GetOptions{}) // need to re-add workers only
```

Comment:
GetOptions won't take a LabelSelector like ListOptions does (maybe there's some way to specify), but since we are using `request.Name` and watching for MachineSet changes, the Get should only be running against individual MachineSet objects if/when they are modified and not the whole list.